cmd/hashsum/hashsum.go

```diff
@@ -55,6 +59,32 @@ func GetHashsumOutput(filename string) (out *os.File, close func(), err error) {
 	return out, close, nil
 }
 
+// HashStream produces hash from a stream
+func HashStream(ht hash.Type, outputBase64 bool, in io.ReadCloser) error {
+	hasher, err := hash.NewMultiHasherTypes(hash.NewHashSet(ht))
+	if err != nil {
+		return errors.Wrap(err, "Hash unsupported")
+	}
+	written, err := io.Copy(hasher, in)
+	fs.Debugf(nil, "Creating %s hash of %d bytes read from input stream", ht, written)
+	if err != nil {
+		return errors.Wrap(err, "Failed to copy input to hasher")
+	}
+	byteSum, err := hasher.Sum(ht)
+	if err != nil {
+		return errors.Wrap(err, "Hasher returned an error")
+	}
+	sum := hex.EncodeToString(byteSum)
```

Comment:
This printing logic is duplicated from operations.go isn't it?

https://github.com/rclone/rclone/blob/b323bf34e2313f692dd013921bf87079bea73bf8/fs/operations/operations.go#L1031-L1040

I wonder whether we should factor that logic into a function and call it here.

The logic is probably complicated enough and using syncPrintf is probably a good idea as the other printing in this routine goes through there.

Reply:
My reasoning for not using syncPrintf was that since this is a "standalone" operation, not part of listing, no stats, or anything (like the original hashing), it was not necessary with syncPrintf.